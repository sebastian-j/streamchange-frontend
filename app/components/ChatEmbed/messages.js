/*
 * ChatEmbed Messages
 *
 * This contains all the text for the ChatEmbed component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'streamchange.components.ChatEmbed';

export default defineMessages({
  panelTitle: {
    id: `${scope}.title`,
    defaultMessage: 'Chat',
  },
});
