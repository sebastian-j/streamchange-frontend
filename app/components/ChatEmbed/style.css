.gv-column {
    background-color: white;
    flex-basis: 0;
    flex-grow: 1;
    margin: 15px;
    padding: 15px;
}
.chat-frame {
    height: calc(100% - 67px);
    width: 100%;
}
