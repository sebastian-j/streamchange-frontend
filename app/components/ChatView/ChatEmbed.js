import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ChatFrame = styled.iframe`
  border: none;
  height: calc(100% - 67px);
  width: 100%;
`;

function ChatEmbed(props) {
  return (
    <ChatFrame
      className="chat-frame"
      title="Youtube Chat"
      src={`https://www.youtube.com/live_chat?v=${props.videoId}&embed_domain=${
        window.location.hostname
      }`}
    />
  );
}

ChatEmbed.propTypes = {
  videoId: PropTypes.string,
};

export default ChatEmbed;
