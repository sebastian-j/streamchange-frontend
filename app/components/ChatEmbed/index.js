import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';

import Panel from '../Panel';
import PanelTitle from '../Panel/PanelTitle';
import messages from './messages';

const ChatFrame = styled.iframe`
  border: none;
  height: calc(100% - 67px);
  width: 100%;
`;

function ChatEmbed(props) {
  return (
    <Panel>
      <PanelTitle>
        <FormattedMessage {...messages.panelTitle} />
      </PanelTitle>
      <ChatFrame
        className="chat-frame"
        title="Youtube Chat"
        src={`https://www.youtube.com/live_chat?v=${
          props.videoId
        }&embed_domain=${window.location.hostname}`}
      />
    </Panel>
  );
}

ChatEmbed.propTypes = {
  videoId: PropTypes.string,
};

export default ChatEmbed;
