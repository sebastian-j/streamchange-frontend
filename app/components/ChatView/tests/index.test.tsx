import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import ChatEmbed from '../ChatEmbed';

const shallowRenderer = createRenderer();

describe('<ChatEmbed />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(<ChatEmbed videoId="id" />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
