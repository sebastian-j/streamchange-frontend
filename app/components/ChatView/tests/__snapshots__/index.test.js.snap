// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatEmbed /> should render and match the snapshot 1`] = `
<ForwardRef
  className="chat-frame"
  src="https://www.youtube.com/live_chat?v=id&embed_domain=localhost"
  title="Youtube Chat"
/>
`;
